src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -92,85 +99,992 @@ static ColliderJntSphInit D_80AD2790 = {
+    if (this->seekPlayerTimer <= 0) {
+        EnPeehat_Ground_SetStateLanding(this);
+        this->riseDelayTimer = 40;
+    } else {
+        this->seekPlayerTimer--;
+    }
+    if ((gSaveContext.nightFlag == 0) &&
+        (Math_Vec3f_DistXZ(&this->actor.home.pos, &player->actor.world.pos) < this->xzDistMax)) {
+        Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 1, 1000, 0);
+        if (this->unk2FA != 0) {
+            this->actor.shape.rot.y += 0x1C2;
+        } else {
+            this->actor.shape.rot.y -= 0x1C2;
+        }
+    } else {
+        EnPeehat_Ground_SetStateReturnHome(this);
+    }
+    SkelAnime_Update(&this->skelAnime);
+    Math_SmoothStepToS(&this->bladeRotVel, 4000, 1, 500, 0);
+    this->bladeRot += this->bladeRotVel;
+    Math_SmoothStepToF(&this->scaleShift, 0.075f, 1.0f, 0.005f, 0.0f);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_PIHAT_FLY - SFX_FLAG);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80ACFB34.s")
+void EnPeehat_Larva_SetStateSeekPlayer(EnPeehat* this) {
+    Animation_PlayLoop(&this->skelAnime, &D_060005C4);
+    this->state = PEAHAT_STATE_SEEK_PLAYER;
+    this->unk2D4 = 0;
+    EnPeehat_SetActionFunc(this, EnPeehat_Larva_StateSeekPlayer);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80ACFBCC.s")
+void EnPeehat_Larva_StateSeekPlayer(EnPeehat* this, GlobalContext* globalCtx) {
+    f32 speedXZ = 5.3f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80ACFD20.s")
+    if (this->actor.xzDistToPlayer <= 5.3f) {
+        speedXZ = this->actor.xzDistToPlayer + 0.0005f;
+    }
+    if (this->actor.parent != NULL && this->actor.parent->update == NULL) {
+        this->actor.parent = NULL;
+    }
+    this->actor.speedXZ = speedXZ;
+    if (this->actor.world.pos.y - this->actor.floorHeight >= 70.0f) {
+        Math_SmoothStepToF(&this->actor.velocity.y, -1.3f, 1.0f, 0.5f, 0.0f);
+    } else {
+        Math_SmoothStepToF(&this->actor.velocity.y, -0.135f, 1.0f, 0.05f, 0.0f);
+    }
+    if (this->unk2D4 == 0) {
+        Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 1, 830, 0);
+    } else {
+        this->unk2D4--;
+    }
+    this->actor.shape.rot.y += 0x15E;
+    SkelAnime_Update(&this->skelAnime);
+    Math_SmoothStepToS(&this->bladeRotVel, 4000, 1, 500, 0);
+    this->bladeRot += this->bladeRotVel;
+    Math_SmoothStepToF(&this->scaleShift, 0.075f, 1.0f, 0.005f, 0.0f);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_PIHAT_SM_FLY - SFX_FLAG);
+    if (this->colQuad.base.atFlags & AT_BOUNCED) {
+        this->actor.colChkInfo.health = 0;
+        this->colQuad.base.acFlags = this->colQuad.base.acFlags & ~AC_BOUNCED;
+        EnPeehat_SetStateAttackRecoil(this);
+        return;
+    }
+    if ((this->colQuad.base.atFlags & AT_HIT) || (this->colCylinder.base.acFlags & AC_HIT) ||
```

Comment:
```suggestion
    } else if ((this->colQuad.base.atFlags & AT_HIT) || (this->colCylinder.base.acFlags & AC_HIT) ||
```

Reply:
idk about this. i feel like using simply an if statement is cleaner here.

Reply:
I agree with petrie here. the checks are all related, its collision flag checks

Reply:
it's not about the relationship of the checks, it's about not needing to scan down 32 lines into the bottom of the next else case to understand that nothing else happens after you go into quad AT_BOUNCE case. But I'll change it anyway